internal/network/ports.go

```diff
@@ -0,0 +1,197 @@
+			return []factory.PortProtocol{{Port: port.Port, Protocol: port.Protocol}}
+		}
+	}
+
+	// Fall back to default port with TCP protocol
+	if defaultPort := getDefaultPort(output.Type, urlStr); defaultPort > 0 {
+		return []factory.PortProtocol{{Port: defaultPort, Protocol: corev1.ProtocolTCP}}
+	}
+
+	return []factory.PortProtocol{}
+}
+
+// getKafkaBrokerPortProtocols extracts ports with protocols from all Kafka brokers.
+func getKafkaBrokerPortProtocols(brokers []obs.BrokerURL) []factory.PortProtocol {
+	portProtocolSlice := []factory.PortProtocol{}
+
+	// Check all broker URLs
+	for _, broker := range brokers {
+		if port := parsePortProtocolFromURL(string(broker)); port != nil {
+			portProtocolSlice = append(portProtocolSlice, factory.PortProtocol{Port: port.Port, Protocol: port.Protocol})
+		} else {
+			// If no port in broker URL, use default port based on scheme
+			if defaultPort := getDefaultPort(obs.OutputTypeKafka, string(broker)); defaultPort > 0 {
+				portProtocolSlice = append(portProtocolSlice, factory.PortProtocol{Port: defaultPort, Protocol: corev1.ProtocolTCP})
+			}
+		}
+	}
+
+	return portProtocolSlice
+}
+
+// parsePortProtocolFromURL extracts the port from a URL string.
+// Returns nil if the port cannot be determined.
+func parsePortProtocolFromURL(urlStr string) *factory.PortProtocol {
+	parsedURL, err := url.Parse(urlStr)
+	if err != nil {
+		return nil
+	}
+
+	// Determine protocol from scheme
+	var protocol corev1.Protocol
+	switch parsedURL.Scheme {
+	case "udp":
+		protocol = corev1.ProtocolUDP
+	default:
+		// Default to TCP
+		protocol = corev1.ProtocolTCP
+	}
+
+	portStr := parsedURL.Port()
+	if portStr != "" {
+		port, err := strconv.ParseInt(portStr, 10, 32)
+		if err == nil && port > 0 {
+			return &factory.PortProtocol{Port: int32(port), Protocol: protocol}
+		}
+	}
+
+	return nil
+}
+
+// getDefaultPort returns the default port for a given output type based on the URL scheme or the default port for the output type.
+func getDefaultPort(outputType obs.OutputType, urlStr string) int32 {
+	// Parse URL to determine scheme for kafka and http/https to return appropriate default port
+	var scheme string
+	if urlStr != "" {
+		if parsedURL, err := url.Parse(urlStr); err == nil {
+			scheme = parsedURL.Scheme
+		}
+	}
+
+	switch outputType {
+	case obs.OutputTypeElasticsearch:
+		return 9200
+	case obs.OutputTypeSplunk:
+		return 8088
+	case obs.OutputTypeLoki:
+		return 3100
+	case obs.OutputTypeSyslog:
+		return 514
+	case obs.OutputTypeOTLP:
+		return 4318
+	case obs.OutputTypeLokiStack: // LokiStack uses 8080
+		return 8080
+	case obs.OutputTypeCloudwatch, obs.OutputTypeAzureMonitor, obs.OutputTypeGoogleCloudLogging:
+		return 443
+	case obs.OutputTypeKafka:
+		// Kafka uses 9092 for plaintext (tcp), 9093 for TLS
+		if scheme == "tls" {
+			return 9093
+		}
+		return 9092
+	case obs.OutputTypeHTTP:
+		if scheme == "http" {
+			return 80
+		}
+		return 443 // https
+	}
+	return 0
```

Comment:
I wonder if we should insert a "panic" here because we should only get here if there is an error in the logic or we missed a place to implement something for a new output